### ISSUE TITLE
Not working with docker-machine 0.5.0

### ISSUE THREAD

**Issue by marclennox** [codekitchen/dinghy#100]
It seems dinghy is not working with docker-machine 0.5.0 which was recently released.  It appears to be a problem with ssh client credentials.

`dinghy create` results in the following:

```
Creating the dinghy VM...
Machine "dinghy" is already running.
/usr/local/Cellar/dinghy/4.0.5/cli/dinghy/machine.rb:27:in `up': There was an error bringing up the VM. Dinghy cannot continue. (RuntimeError)
	from /usr/local/Cellar/dinghy/4.0.5/cli/cli.rb:184:in `start_services'
	from /usr/local/Cellar/dinghy/4.0.5/cli/cli.rb:54:in `create'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor/command.rb:27:in `run'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor/invocation.rb:126:in `invoke_command'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor.rb:359:in `dispatch'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor/base.rb:440:in `start'
	from /usr/local/bin/_dinghy_command:6:in `<main>'
```

The vm gets created and started but `dinghy status` shows

```
  VM: running
 NFS: running
FSEV: not running
 DNS: not running
HTTP: not running
```

After that, I can ssh just fine with `docker-machine ssh dinghy`

But `dinghy ssh` results in `Permission denied (publickey,password,keyboard-interactive).`

**Comment by marclennox** [comment]
I noticed that the `~/.dinghy/ssh-config` file is missing path information for the ssh key

```
Host dinghy
  HostName 192.168.99.100
  User docker
  Port 22
  UserKnownHostsFile /dev/null
  StrictHostKeyChecking no
  PasswordAuthentication no
  IdentityFile /id_rsa
  IdentitiesOnly yes
  LogLevel ERROR
```

After manually editing the IdentityFile path, I'm able to `dinghy ssh` ok, but `dinghy up` resets it to the wrong path

```
Starting the dinghy VM...
Starting NFS daemon, this will require sudo
Waiting for NFS daemon...
Mounting NFS /Users/Marc
Permission denied (publickey,password,keyboard-interactive).
/usr/local/Cellar/dinghy/4.0.5/cli/dinghy/ssh.rb:23:in `run': Error executing command: ["sudo umount /Users || true"] (Ssh::CommandFailed)
	from /usr/local/Cellar/dinghy/4.0.5/cli/dinghy/machine.rb:76:in `ssh'
	from /usr/local/Cellar/dinghy/4.0.5/cli/dinghy/machine.rb:69:in `mount'
	from /usr/local/Cellar/dinghy/4.0.5/cli/cli.rb:186:in `start_services'
	from /usr/local/Cellar/dinghy/4.0.5/cli/cli.rb:78:in `up'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor/command.rb:27:in `run'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor/invocation.rb:126:in `invoke_command'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor.rb:359:in `dispatch'
	from /usr/local/Cellar/dinghy/4.0.5/cli/thor/lib/thor/base.rb:440:in `start'
	from /usr/local/bin/_dinghy_command:6:in `<main>'
```

**Comment by neilgupta** [comment]
Can confirm I was seeing the same issue, running 01feb65 from Dinghy `master`. I had to downgrade to docker-machine 0.4.1 to get it working again.

**Comment by marclennox** [comment]
I was able to get the machine running by manually creating it with docker-machine

`docker-machine create dinghy`

Then creating a symbolic link (hack alert) as follows, with proper permissions

`/id_rsa -> ~/.docker/machine/machines/dinghy/id_rsa`

**Comment by codekitchen** [comment]
I've released dinghy 4.0.6 with docker-machine 0.5.0 compatibility. Thanks @miquella for the help while I was out camping in the backcountry.

**Comment by chinthakagodawita** [comment]
Hey @codekitchen, the initial proposed change in #102 works for me but `v4.0.6` doesn't.

It seems that just the `StorePath` property doesn't include the full path to the identity file, you need to append the word `machines` and the name of the machine.

i.e. the path in 4.0.6 - `/Users/chin/.docker/machine/id_rsa` doesn't exist, however the path generated in the change in #102 - `/Users/chin/.docker/machine/machines/dinghy/id_rsa` does exist and works.

**Comment by codekitchen** [comment]
Can you paste the output of running `docker-machine inspect dinghy` on your mac? That's not what I'm seeing locally with docker-machine version 0.5.0 installed from homebrew. `.HostOptions.AuthOptions.StorePath` in the json output points directly to `/Users/codekitchen/.docker/machine/machines/dinghy`.

**Comment by chinthakagodawita** [comment]
Sure, it's over at https://gist.github.com/chinthakagodawita/a600464bded9401fcc65

I did upgrade to docker/boot2docker 1.9 recently, not sure if that's causing this.

**Comment by codekitchen** [comment]
Geez yeah, it looks like the value can be different if you've upgraded from an older docker-machine release. I originally tested my change on a newly imaged machine, but testing it on my daily dev machine I got the same output as you see at first -- until I dropped and re-created the VM, and then I get the full path.

I don't have time to dig into the docker-machine code right now to see why that's the case, so I've reverted that out for now. Dinghy 4.0.7 should work for you.
